### DISCUSSION TITLE
How to make installation easier?

### DISCUSSION THREAD

**Discussion by markseuffert** [datenstrom/community#9] [Ask a question] [answered]
I think the installation is a problem right now, it takes too much time to get started. I did this for some friends on a simple web hosting, just FTP access without SSH/Dropbox. You have to edit system configuration, switch to http when SSL is not available (sigh), add an user account to the web interface, etc... before you can start editing pages in the browser. 

What can be done to make installation easier?

**Comment by ofisia** [comment]
- Set `webinterfaceServerScheme = http` as the default.
- Set default username and password.
- Include [password generator](https://demo.datenstrom.se/tests/new-user) in the default installation.

**Comment by markseuffert** [comment]
Good idea. Is it possible to avoid a default username and password? Not that strangers can log in. How about a form, that you fill out when you open your website the first time? For example your name, email and password. It would be one additional installation step:

1. Download Yellow and unzip it.
2. Copy all files to your web hosting.
3. Open your website in a browser **and complete the installation.**

There has to be an automatic trouble shooting:
- check PHP version 5.3 or later
- check web server is known
- check web server supports rewrite 
- check access rights (not sure, not everyone needs write access for web interface)

**Comment by dragonbane** [comment]
Is this something you're looking at?  If not, I would like to help.

**Comment by markseuffert** [comment]
I haven't started on it. Help is much appreciated... or if you want we can [do it together](https://datenstrom.se/yellow/help/contributing-guidelines).

**Comment by gosteev** [comment]
hello. i've got some problems during installation. by default users want to see some basic example after installation, but i have only one non-index .php file and if i understood correctly, should make some magic configuration to make website work :)

maybe there should be more info on "how to install"? because i just followed instructions, and get lost.
or publish your demo website http://demo.datenstrom.se/ as repository, it will be great solution

**Comment by markseuffert** [comment]
Hi there, what's the problem during Yellow installation?

**Comment by gosteev** [comment]
thank you :)

well, i upload files via ftp, go to yellow.php and...:
![2014-12-12 13-21-39 file not found - yellow](https://cloud.githubusercontent.com/assets/6203574/5410136/8b1bb282-8202-11e4-9d4e-f79275d36bfe.png)

**Comment by markseuffert** [comment]
Can you tell me which web server you have?

**Comment by gosteev** [comment]
apache 2. can't tell version exactly

**Comment by markseuffert** [comment]
It could be one of two things that's causing the problem. Either the `.htaccess` file is missing or URL rewrite is not enabled on your web server. Sorry about the trouble. Looks like we should add a self-check to the Yellow installation.

**Comment by gosteev** [comment]
got it. `.htaccess` file exists, i will look towards url rewrite, thanks :)

**Comment by markseuffert** [comment]
I don't have time to do an installer at the moment. Would be good to have though.

**Comment by markseuffert** [comment, marked as answer]
[Installer](https://github.com/annaesvensson/yellow-install) has been added to latest release. 😄
